packages/core/test/util/txInspector.test.ts

```diff
@@ -405,4 +577,56 @@ describe('txInspector', () => {
       }
     );
   });
+
+  describe('mint and brun transaction inspector', () => {
+    test('inspects a transaction that mints and burns tokens and can retrieve the minting details', () => {
+      const tx = buildMockTx({ includeAuxData: true });
+      const inspectTx = createTxInspector({ assetsMinted: assetsMintedInspector });
+      const { assetsMinted } = inspectTx(tx);
+
+      expect(assetsMinted.minted.length).toEqual(2);
+      expect(assetsMinted.burned.length).toEqual(1);
+      expect(assetsMinted.minted[0].assetName).toEqual(mockTokenName1);
+      expect(assetsMinted.minted[0].policyId).toEqual(mockPolicy1);
+      expect(assetsMinted.minted[0].fingerprint).toEqual(
+        Cardano.AssetFingerprint.fromParts(Cardano.PolicyId(mockPolicy1), Cardano.AssetName(mockTokenName1))
+      );
+      expect(assetsMinted.minted[0].quantity).toEqual(1n);
+      expect(assetsMinted.minted[0].script).toEqual(mockScript1);
+
+      expect(assetsMinted.minted[1].assetName).toEqual(mockTokenName2);
+      expect(assetsMinted.minted[1].policyId).toEqual(mockPolicy2);
+      expect(assetsMinted.minted[1].fingerprint).toEqual(
+        Cardano.AssetFingerprint.fromParts(Cardano.PolicyId(mockPolicy2), Cardano.AssetName(mockTokenName2))
+      );
+      expect(assetsMinted.minted[1].quantity).toEqual(100_000n);
+      expect(assetsMinted.minted[1].script).toEqual(mockScript2);
+
+      expect(assetsMinted.burned[0].assetName).toEqual(mockTokenName3);
```

Comment:
This reads a little awkward. Maybe we should either:
a) change the inspector to just return a single array (instead of `{minted, burned}`) and document that negative quantity is burning?
b) split it into 2 inspectors, so that you could `createTxInspector({ mint, burn })`

Reply:
Agree, I like both options, maybe option B is a bit better?

Reply:
addressed here https://github.com/input-output-hk/cardano-js-sdk/pull/447/commits/c0be40bf0295b92a6b26bd52de82e9baba42d527